### PR TITLE
Minor edits to let-it-throw

### PR DESCRIPTION
**Pull request by phatblat**
> These changes are merely suggestions. Feel free to take from them what you want, or ignore them completely.

This article is fantastic! I'm a huge fan of the Frozen ❄️ puns and think it makes a rather dry topic more interesting. One thought: _add frozen images with your Swift-ified quotes embedded as memes?_ Could help break up the text without adding more code samples. I really like how you touch on the fact that `NSError` is just another `ErrorType` with concrete examples. That's a key tidbit for migration that I haven't seen covered much at all. Also, I ❤️ the abstraction example. That's a hard area for people to get sometimes and it really helps to have a good example like this.

On the edits, I'm coming from an American English perspective so things like "happy path" are just how I've heard it the most. But "good path" certainly communicates the meaning, so that is fine, especially if that's a more common term across the pond.

Great job!

### CODE REVIEW

File: _drafts/let-it-throw.md [hunk 1]
```diff
@@ -137,32 +137,32 @@ func doTheActualCall() throws {
...
+1. Your users shouldn't have to know which internal library you're using. If some day in the future you decide to switch your implementation to use `SomeOtherPopularLibA` instead of `MyLibA`, which will obviously not throw exactly the same errors, the caller of your own `MyLibC` framework shouldn't need to know or care. That's what abstraction is all about.
+1. The caller shouldn't have to handle all the errors. Surely you can catch some of those errors and consider them internal: not all errors thrown by `MyLibA` make sense to be exposed publicly to your users, for example a `FrameworkConfigurationError` error mentioning that you misused the `MyLibA` framework and forgot to call its `setup()` method or whatever should not make its way to the user, as the user can't do much about it. That kind of error is your fault, not his/hers.
```
Comment (AliSoftware) [inline _drafts/let-it-throw.md:163]: Let's use `theirs`, even better gender-neutral

Comment (phatblat) [inline _drafts/let-it-throw.md:163]: :+1: even better. I hate how English doesn't have a gender-neutral pronoun. "y'all" is the best we have.

**Comment by AliSoftware** [comment]
Thanks a lot for the review! :ok_hand: Really appreciate the feedback.

> add frozen images with your Swift-ified quotes embedded as memes?

Good idea :+1:  will do that in the train tomorrow morning

**Comment by AliSoftware** [comment]
> Also, I ❤️ the abstraction example. That's a hard area for people to get sometimes and it really helps to have a good example like this.

I was actually wondering about using a more concrete example as I find that paragraph… a bit abstract, ironically :smile: like use more concrete examples of library names and error types instead of `MyLibA`, `MyLibAError` etc? Do you think it's worth trying to find a matching example, or maybe it's better to keep it abstract?

**Comment by phatblat** [comment]
I like the simplicity of your current approach. It's easy to get lost in A, B & C when the reader came to learn about Swift `throws`, but you make a solid point of **don't burden your users with implementation details**.

**Comment by phatblat** [comment]
I like the A/B/C because you show how `MyLibA` could be replaced with `SomeOtherPopularLibA`. But to reduce the alphabet-soup noise you could use examples like `AFNetworking` and `AlamoFire`.

**Comment by AliSoftware** [comment]
Will probably keep it that way then ;)

btw if you have links to interesting blog articles about error, throw, rethrow and all, I'm all ears. I want to link to more than just Rob's one.

**Comment by phatblat** [comment]
I'm always good for a list o' links:
- https://littlebitesofcocoa.com/108-error-handling
- https://realm.io/news/testing-swift-error-type
- http://www.splinter.com.au/2015/06/21/swift-keychain-wrapper
- https://gist.github.com/nicklockwood/21495c2015fd2dda56cf
- https://www.hackingwithswift.com/new-syntax-swift-2-error-handling-try-catch
- http://www.sunsetlakesoftware.com/2015/06/12/swift-2-error-handling-practice
- http://www.sunsetlakesoftware.com/2015/11/03/what-we-learned-rewriting-our-robotic-control-software-swift
- http://nshipster.com/nserror/

**Comment by AliSoftware** [comment]
Yeah I actually thought about AFN vs. Alamofire but then started to think that some would argue in the comments that the higher layer should be aware of all AFN errors like HTTP codes and all so that hiding them would not be a good practice (then I'll have to explain them that I didn't mean them to hide them but just rethrow them with another, more abstract type, etc… didn't want to explain that all in the post which was quite long already :wink:

That's one of the hardest part of blogging I think, both being concise to avoid loosing the reader, but also please the nitpickers and people wanting to go deeper. I first intended my blog to be about higher level topics, like not re-explain the basics of Swift which are explained in Apple's iBook but more practical ways to make great use of them, like I did with my articles about enums and Mixins. But I realize for some less-known topics like this one we have to bring the basics first.

**Comment by AliSoftware** [comment]
Thx for the links, will add them :+1: 
The one in NSHipster I'm not so sure, that's quite ObjC-oriented only, but the other two will probably be a great addition

**Comment by phatblat** [comment]
Hrm, yeah. Concrete examples could be distracting.

I just included the NSHipster link so you could consider linking to it for Swift beginners that haven't worked with @objc before.

**Comment by phatblat** [comment]
Other "further reading" types of links that I don't have handy: `try` and `rethrows`. Sure, people can google for those, but I like to provide a big list of references at the end of my (few) talks and plan to start doing this on posts as well.

**Comment by AliSoftware** [comment]
Yeah actually:
- The `rethrows` is why I linked Rob's article at first.
- I plan my next article to be a follow-up of this one, talking about `try!` and `try?`, error handling in closures, `rethrows`, and `Result` — which I intended to talk about in this current article (introducing it as "before `throws` we had to use `Result` blah blah)

But as the article was getting too long already, I split it and [kept those paragraphs for the next draft](https://github.com/AliSoftware/alisoftware.github.io/blob/master/_drafts/result.md) :wink:
Might want to add more links about `rethrows` by that time

**Comment by AliSoftware** [comment]
[added](https://github.com/AliSoftware/alisoftware.github.io/compare/e45604461e57c2dc78e3895e52a3591a873604cd...master) your links, thx again!

**Pull request merged**